File: LcWithPatterns/JavaScript/HashMap/lc219-ContainsDuplicate2.js
"use strict";
function containsNearbyDuplicate(nums, k) {
    const map = new Map();
    for (let i = 0; i < nums.length; i++) {
        const currentNum = nums[i];
        if (map.has(currentNum) && i - map.get(currentNum) <= k) {
            return true;
        }
        map.set(currentNum, i);
    }
    return false;
}

File: LcWithPatterns/Python/HashMap/lc219-ContainsDuplicate2.py
from typing import List

def containsNearbyDuplicate(nums: List[int], k: int) -> bool:
    num_map = {}

    for i, currentNum in enumerate(nums):
        if currentNum in num_map and i - num_map[currentNum] <= k:
            return True

        num_map[currentNum] = i

    return False
